lib/cocina/generator/generator.rb

```diff
@@ -100,9 +100,13 @@ def schema_for(schema_name)
 
         case schema_doc.type
         when 'object'
-          Schema.new(schema_doc)
+          Schema.new(schema_doc, schemas: schemas.keys)
         when 'string'
-          Datatype.new(schema_doc)
+          Datatype.new(schema_doc, schemas: schemas.keys)
```

Comment:
These changes allow all schemas to know the entirety of the named schemas from the API doc so that they can reference one another.